src/CommunityToolkit.Maui.MediaElement/MediaElement.shared.cs

```diff
@@ -0,0 +1,502 @@
+		set => SetValue(SpeedProperty, value);
+	}
+
+	/// <summary>
+	/// Gets the height (in pixels) of the loaded media in pixels.
+	/// This is a bindable property.
+	/// </summary>
+	/// <remarks>Not reported for non-visual media.</remarks>
+	public int VideoHeight
+	{
+		get => (int)GetValue(VideoHeightProperty);
+		internal set => SetValue(VideoHeightProperty, value);
+	}
+
+	/// <summary>
+	/// Gets the width (in pixels) of the loaded media in pixels.
+	/// This is a bindable property.
+	/// </summary>
+	/// <remarks>Not reported for non-visual media.</remarks>
+	public int VideoWidth
+	{
+		get => (int)GetValue(VideoWidthProperty);
+		internal set => SetValue(VideoWidthProperty, value);
+	}
+
+	/// <summary>
+	/// The current state of the <see cref="MediaElement"/>. This is a bindable property.
+	/// </summary>
+	public MediaElementState CurrentState
+	{
+		get => (MediaElementState)GetValue(CurrentStateProperty);
+		private set => SetValue(CurrentStateProperty, value);
+	}
+
+	TimeSpan IMediaElement.Position
+	{
+		get => (TimeSpan)GetValue(PositionProperty);
+		set
+		{
+			var currentValue = (TimeSpan)GetValue(PositionProperty);
+
+			if (currentValue != value)
+			{
+				SetValue(PositionProperty, value);
+				OnPositionChanged(new(value));
+			}
+		}
+	}
+
+	TimeSpan IMediaElement.Duration
+	{
+		get => (TimeSpan)GetValue(DurationProperty);
+		set => SetValue(durationPropertyKey, value);
+	}
+
+	/// <inheritdoc cref="IMediaElement.Pause"/>
+	public void Pause()
+	{
+		OnPauseRequested();
+		Handler?.Invoke(nameof(PauseRequested));
+	}
+
+	/// <inheritdoc cref="IMediaElement.Play"/>
+	public void Play()
+	{
+		InitializeTimer();
+		OnPlayRequested();
+		Handler?.Invoke(nameof(PlayRequested));
+	}
+
+	/// <inheritdoc cref="IMediaElement.SeekTo(TimeSpan)"/>
+	public void SeekTo(TimeSpan position)
+	{
+		MediaSeekRequestedEventArgs args = new(position);
+		Handler?.Invoke(nameof(SeekRequested), args);
+	}
+
+	/// <inheritdoc cref="IMediaElement.Stop"/>
+	public void Stop()
+	{
+		ClearTimer();
+		OnStopRequested();
+		Handler?.Invoke(nameof(StopRequested));
+	}
+
+	internal void OnMediaEnded()
+	{
+		ClearTimer();
+		CurrentState = MediaElementState.Stopped;
+		eventManager.HandleEvent(this, EventArgs.Empty, nameof(MediaEnded));
+	}
+
+	internal void OnMediaFailed(MediaFailedEventArgs args)
+	{
+		((IMediaElement)this).Duration = ((IMediaElement)this).Position = TimeSpan.Zero;
+
+		CurrentState = MediaElementState.Failed;
+		eventManager.HandleEvent(this, args, nameof(MediaFailed));
+	}
+
+	internal void OnMediaOpened()
+	{
+		InitializeTimer();
+		eventManager.HandleEvent(this, EventArgs.Empty, nameof(MediaOpened));
+	}
+
+	/// <inheritdoc/>
+	protected override void OnBindingContextChanged()
+	{
+		if (Source != null)
+		{
+			SetInheritedBindingContext(Source, BindingContext);
+		}
+
+		base.OnBindingContextChanged();
+	}
+
+	static void OnAutoPlayPropertyChanged(BindableObject bindable, object oldValue, object newValue) =>
```

Comment:
Just out of curiosity, why we aren't using mapper for those properties instead?

Reply:
Old habits die hard...